app/lib/search/backend.dart

```diff
@@ -60,9 +61,139 @@ void registerPackageIndex(PackageIndex index) =>
 /// Datastore-related access methods for the search service
 class SearchBackend {
   final DatastoreDB _db;
+  final VersionedJsonStorage _snapshotStorage;
   final _http = httpRetryClient();
 
-  SearchBackend(this._db);
+  SearchBackend(this._db, Bucket snapshotBucket)
+      : _snapshotStorage = VersionedJsonStorage(snapshotBucket, 'snapshot/');
+
+  /// Runs a forever loop and tries to get a global lock.
+  ///
+  /// Once it has the claim, it creates a new snapshot by loading
+  /// all visible package data and uploads it into the snapshot storage.
+  /// Tracks the package updates for the next up-to 24 hours and writes
+  /// the snapshot after every 10 minutes.
+  ///
+  /// When other process has the claim, the loop waits a minute before
+  /// attempting to get the claim.
+  Future<void> updateSnapshotInForeverLoop() async {
+    final lock = GlobalLock.create(
+      'update-search-snapshot',
+      expiration: Duration(minutes: 20),
+    );
+    for (;;) {
+      try {
+        await lock.withClaim((claim) async {
+          await doCreateAndUpdateSnapshot(claim);
+        });
+      } catch (e, st) {
+        _logger.warning('Snapshot update failed.', e, st);
+      }
+      await Future.delayed(Duration(minutes: 1));
```

Comment:
```suggestion
      // Wait for 1 minutes for sanity, before trying again.
      await Future.delayed(Duration(minutes: 1));
```

I assume this is just for sanity? to avoid a run away loop?